snakemake/__init__.py

```diff
@@ -1923,6 +1927,32 @@ def get_argument_parser(profile=None):
         "Currently slack and workflow management system (wms) are supported.",
     )
 
+    group_slurm = parser.add_argument_group("SLURM")
+    slurm_mode_group = group_slurm.add_mutually_exclusive_group()
+
+    slurm_mode_group.add_argument(
+        "--slurm",
+        action="store_true",
+        help=(
+            "Execute snakemake rules as SLURM batch jobs according"
+            " to their 'resources' definition. SLRUM resources as "
+            " 'partition', 'ntasks', 'cpus', etc. need to be defined"
+            " per rule within the 'resources' definition. Note, that"
+            " memory can only be defined as 'mem_mb' or 'mem_mb_per_cpu'"
+            " as analoguous to the SLURM 'mem' and 'mem-per-cpu' flags"
+            " to sbatch, respectively. Here, the unit is always 'MiB'."
```

Comment:
hm, I will do so, but first, please answer/comment this, first: All RAM is bought in MiB, storage sold in MB (or bigger scales). Should a workflow system which intends to allocate RAM not always handle the binary prefix?

Reply:
MB is more widely known than MiB, hence, defining mem_mb instead of mem_mib in rule resources is more readable to non-technical people.

Reply:
yep, let me re-phrase: Should snakemake not do the conversion _always_, like "dear user, you / we write MB, but really mean MiB"? For, the upper limit of a system, not withstanding OS and service daemons requirements, cannot be reached when submitting with MB. If we do the conversion for this reason in the SLURM executor only, then this is an exception to a rule, which applies everywhere - albeit less explicitly.

Yet, I do understand hesitation at this point, as this would mean a global change to snakemakes established behaviour.

Reply:
Good point, but then maybe just let us convert automatically. Whenever the `mem_mb` resource is defined, the job automatically adds a `mem_mib` resource with the appropriate conversion.

Reply:
I will take care of that in a separate PR.

Reply:
Does that make sense?